test/compress/reduce_vars.js

```diff
@@ -2417,3 +2432,32 @@ issue_1850_4: {
     }
     expect_stdout: true
 }
+
+issue_1865: {
+    options = {
+        evaluate: true,
+        reduce_vars: true,
+        unsafe: true,
+    }
+    input: {
+        function f(a) {
+            a.b = false;
```

Comment:
Grrr... of course I forgot to change this.

I turned `a.b` from the original example into `some.thing` so I don't need to rely on `booleans`, which turns `true` into `!0` in order for the smart variable replacement logic to kick in.